### PR TITLE
PE-19 | Implement First Time Onboarding

### PR DESCRIPTION
**Pull request by dominikyz**
**What I did:**

- Created First time onboarding page

**How to test:**

- Clone branch
- swipe right/left or click small dots to see different slides
- test skip button
- test got it button in the end
- upon clicking got it - you should see placeholder page

### CODE REVIEW

**Comment by ayeejane** [comment]
Haven't tested this but make sure to include an explanation of 'What you did' and also 'How to Test' it, so someone reviewing it knows exactly what they should be looking out for.

**Comment by ayeejane** [comment]
And our naming convention for PR's should be of this form `PE-27 | Created side menu` :)

**Comment by dominikyz** [comment]
Hi Jane, comments are fixed, i've also uploaded missing assets folder, so everything should be good to go :)

**Comment by txt020** [comment]
image
Not sure if this is part of your code, but I am getting a weird word or box on my screen.
@dominikyz

File: lib/onboarding.dart [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+class onboarding extends StatefulWidget {
+  @override
+  _onboardingState createState() => _onboardingState();
+}
+
+class _onboardingState extends State<onboarding> {
+  List<PageViewModel> getPages() {
+    return [
+      //-----------------------------------------------------------Welcome
```
Comment (txt020) [inline lib/onboarding.dart]: Can you remove this Welcome Comment

File: lib/onboarding.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      appBar: AppBar(
+        title: Text("Eureka"),
+      ),
+      body: IntroductionScreen(
+        globalBackgroundColor: Colors.white,
+        pages: getPages(),
+        showNextButton: true,
+        showSkipButton: true,
+        skip: Text("Skip"),
+        done: Text("Got it "),
```
Comment (txt020) [inline lib/onboarding.dart]: Can you change these to single quotes

File: lib/onboarding.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        onDone: () {
+          Navigator.push(
+              context,
+              MaterialPageRoute(
+                  builder: (context) =>
+                      placeholder())); // go to placeholder when done is pressed
+        },
```
Comment (txt020) [inline lib/onboarding.dart]: Can you remove this comment

File: lib/onboarding.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        image: Image.asset("assets/images/onboarding2.gif"),
+        title: "Rich collaboration platform",
+        body:
+            "Eureka provides you with every tool to conveniently collaborate on any problem ",
+        decoration: const PageDecoration(
+            //pageColor: Color(0xfffdfafc),
+            ),
+        footer: Text("Choose from Text chat, Video chat, or Live Video Call"),
+      ),
+      //-----------------------------------------------------------Privacy
```
Comment (txt020) [inline lib/onboarding.dart]: Can you change the size font for this footer? It is a little small on my device.
image

File: lib/onboarding.dart [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      //-----------------------------------------------------------Categories
+      PageViewModel(
+        image: Image.asset("assets/images/onboarding1.gif"),
+        title: "Multiple categories",
+        body:
+            "Eureka has several most popular categories for questions and answers",
+        decoration: const PageDecoration(
+          pageColor: Color(0xfffdfafc),
+        ),
+        footer: Text("Lifestyle  Education  Electronics  Household"),
```
Comment (txt020) [inline lib/onboarding.dart]: Here too the footer font size is a little small.
image

**Review by txt020 — CHANGES_REQUESTED** [review]

**Comment by dominikyz** [comment]
> image
> Not sure if this is part of your code, but I am getting a weird word or box on my screen.
> @dominikyz

didnt happen on mine, i'll check whats going on there

File: lib/onboarding.dart (hunk 5 above)
Comment (dominikyz) [reply lib/onboarding.dart]: sure, no problem

File: lib/onboarding.dart (hunk 4 above)
Comment (dominikyz) [reply lib/onboarding.dart]: sure, no problem

File: lib/onboarding.dart (hunk 3 above)
Comment (dominikyz) [reply lib/onboarding.dart]: sure, no problem

File: lib/onboarding.dart (hunk 2 above)
Comment (dominikyz) [reply lib/onboarding.dart]: ok

File: lib/onboarding.dart (hunk 1 above)
Comment (dominikyz) [reply lib/onboarding.dart]: ok

File: lib/onboarding.dart [hunk 6]
```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:introduction_screen/introduction_screen.dart';
+
+class onboarding extends StatefulWidget {
```
Comment (ayeejane) [inline lib/onboarding.dart]: Not that important but change to `Onboarding`

File: lib/onboarding.dart [hunk 7]
```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import 'package:introduction_screen/introduction_screen.dart';
+
+class onboarding extends StatefulWidget {
+  @override
+  _onboardingState createState() => _onboardingState();
+}
+
+class _onboardingState extends State<onboarding> {
```
Comment (ayeejane) [inline lib/onboarding.dart]: Change to `_OnboardingState` and `<Onboarding>`

File: lib/onboarding.dart [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  Widget build(BuildContext context) {
+    return Container(
+      color: Colors.blueAccent,
+      child: Center(
+        child: Text('placeholder'),
+      ),
+    );
+  }
+}
+//placeholder class end
```
Comment (ayeejane) [inline lib/onboarding.dart]: I realized you had created your file under `lib` but for better organization we create different folders https://medium.com/flutter-community/flutter-code-organization-de3a4c219149 This talks about it more in depth but for us, we have a `components` and `screens` folder right now. Please add this file inside `screens` :)

**Review by ayeejane — CHANGES_REQUESTED** [review]

**Comment by dominikyz** [comment]
@ayeejane  all addressed issues have been fixed :)

File: lib/screens/onboarding.dart [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  Widget build(BuildContext context) {
+    return Container(
+      color: Colors.blueAccent,
+      child: Center(
+        child: Text('placeholder'),
+      ),
+    );
+  }
+}
+//placeholder class end
```
Comment (tonynguyen98) [inline lib/screens/onboarding.dart]: When merging, remember to remove this placeholder and replace it with a proper route

**Review by tonynguyen98 — APPROVED** [review]
Great job! Everything look good for your first Flutter application!

**Review by ayeejane — APPROVED** [review]
Looks good to me! Nice job.

**Review by txt020 — APPROVED** [review]
Everything looks good!! Great Job!

**Review by txt020 — APPROVED** [review]

**Pull request merged**